rust/parser/typeql.pest

```diff
@@ -74,15 +74,15 @@ statement = { statement_single | statement_type | statement_thing }
 
 // TYPE STATEMENTS =============================================================
 
-statement_type = { kind ~ type_ref_any ~ ( type_constraint ~ ( COMMA ~ type_constraint )* ~ COMMA? )?
-                 | type_ref_any ~ type_constraint ~ ( COMMA ~ type_constraint )* ~ COMMA?
+statement_type = { kind ~ type_ref ~ ( type_constraint ~ ( COMMA ~ type_constraint )* ~ COMMA? )?
+                 | type_ref~ type_constraint ~ ( COMMA ~ type_constraint )* ~ COMMA?
                  }
 type_constraint = { type_constraint_base ~ annotations? }
 type_constraint_base = { sub_constraint | value_type_constraint | label_constraint
                        | owns_constraint | relates_constraint | plays_constraint
                        }
 
-sub_constraint = { SUB_ ~ type_ref_any }
+sub_constraint = { SUB_ ~ type_ref }
```

Comment:
We can no longer do `$x[] sub myrole[]` etc.

Reply:
Ok!